pkg/api/team.go

```diff
@@ -24,15 +24,22 @@ func (hs *HTTPServer) CreateTeam(c *m.ReqContext, cmd m.CreateTeamCommand) Respo
 	}
 
 	if c.OrgRole == m.ROLE_EDITOR && hs.Cfg.EditorsCanAdmin {
-		addMemberCmd := m.AddTeamMemberCommand{
-			UserId:     c.SignedInUser.UserId,
-			OrgId:      cmd.OrgId,
-			TeamId:     cmd.Result.Id,
-			Permission: m.PERMISSION_ADMIN,
-		}
-
-		if err := hs.Bus.Dispatch(&addMemberCmd); err != nil {
-			c.Logger.Error("Could not add creator to team.", "error", err)
+		// if the request is authenticated using API tokens
+		// the SignedInUser is an empty struct therefore
+		// an additional check whether it is an actual user is required
+		if c.SignedInUser.IsRealUser() {
+			addMemberCmd := m.AddTeamMemberCommand{
+				UserId:     c.SignedInUser.UserId,
+				OrgId:      cmd.OrgId,
+				TeamId:     cmd.Result.Id,
+				Permission: m.PERMISSION_ADMIN,
+			}
+
+			if err := hs.Bus.Dispatch(&addMemberCmd); err != nil {
+				c.Logger.Error("Could not add creator to team.", "error", err)
+			}
+		} else {
+			c.Logger.Error("Could not add creator to team because is not a real user.")
```

Comment:
Only nitpick I have here. @marefr you mentioned in the issue you would expect an error log here, but as nothing breaks and we (almost explicitly) allow API token users to create these, wouldn't it make more sense to be a warning? Not sure if this is a big issue, but from monitoring standpoint if somebody was creating teams programatically this would generate errors even though there is no way around it, is there?

Reply:
Yeah probably warning is better